### PR TITLE
fix: mysql sample delete part

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

File: sample-operators/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaReconciler.java [hunk 1]
```diff
@@ -32,9 +33,9 @@ public MySQLSchemaReconciler() {}
     Secret secret = context.getSecondaryResource(Secret.class).orElseThrow();
     SchemaDependentResource schemaDependentResource = context.managedDependentResourceContext()
         .getDependentResource(SchemaDependentResource.class);
-    return schemaDependentResource.fetchResource(schema).map(s -> {
+    return schemaDependentResource.getResource(schema).map(s -> {
```
Comment (metacosm) [inline sample-operators/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaReconciler.java:36]: Why the change here?

**Review by metacosm — APPROVED** [review]
LGTM

File: sample-operators/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaReconciler.java (hunk 1 above)
Comment (csviri) [reply sample-operators/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaReconciler.java:36]: It's unrelated to the problem. I think we should rather get the actual state from cache, and handle the details when to call the actual service to the dependent resource implementation.
